master/buildbot/process/build.py

```diff
@@ -579,6 +580,14 @@ def getSummaryStatistic(self, name, summary_fn, initial_value=_sentinel):
         else:
             return reduce(summary_fn, step_stats_list, initial_value)
 
+    @defer.inlineCallbacks
+    def postProperties(self):
+        # we get all properties, post to MetricsService to filter out
+        # the one that a user wants to post.
+        properties = interfaces.IProperties(self)
+        yield self.master.stats_service.postProperties(properties,
+                                                       self.builder.name)
+
```

Comment:
mq to write service like statistics service without having to hardocde hook in the code like you do
It works also better with multi master, as you will only need one master to do the stats, while with your approach, you need all master to have a stat service, and make sure they dont step in each other toe.